src/rust/aggregator/src/data/user_history.rs

```diff
@@ -470,10 +467,14 @@ async fn aggregate_change<'a>(
         .await
         .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
     }
+    // Enforce that order status is set to open after a size change to guard against async issues
     sqlx::query!(
         r#"
             UPDATE aggregator.user_history
-            SET remaining_size = $1, last_updated_at = $4
+            SET
+                last_updated_at = $4,
+                order_status = 'open',
```

Comment:
Why would you set order status to open ? It seems not very useful since an order cannot be re-opened. It can also introduce errors if cancel and change events are not emitted or processed in error. I think there is no downside to not setting the status to open so why set it to open ?

Reply:
Consider:

1. User opens order with remaining size 10
2. User changes remaining size to 15
3. 10 fills

If the events are aggregated out of order, (e.g. event 1, then 3, then 2), the order will be marked closed

Reply:
Addressed in 0eb778a